packages/fast-data-utilities-react/src/index.ts

```diff
@@ -675,110 +720,161 @@ export function mapDataToComponent(
+            return mapPluginToData(
+                mappedDataLocation as PluginLocation,
+                data,
+                plugins,
+                childOptions
             );
-            const subData: any = get(mappedData, reactChildrenDataLocation);
-            const isChildString: boolean = typeof subData === "string";
-            const subDataNormalized: any = isChildString
-                ? subData
-                : get(subData, propsKeyword);
-            const childOption: ChildOptionItem = getChildOptionBySchemaId(
-                subSchemaId,
+        case DataResolverType.component:
+        default:
+            return mapDataToChildren(
+                data,
+                mappedDataLocation.absoluteDataLocation,
                 childOptions
             );
+    }
+}
 
-            if (!isChildString) {
-                let value: any;
-
-                if (!childOption) {
-                    value = Object.assign(
-                        { id: subSchemaId },
-                        React.createElement(
-                            React.Fragment,
-                            { key: `${subSchemaId}-${index}` },
-                            subDataNormalized
-                        )
-                    );
-                } else {
-                    value = Object.assign(
-                        { id: subSchemaId },
-                        React.createElement(childOption.component, {
-                            key: `${subSchemaId}-${index}`,
-                            ...subDataNormalized,
-                        })
-                    );
-                }
+function mapPluginToData(
+    pluginModifiedDataLocation: PluginLocation,
+    data: any,
+    plugins: Array<Plugin<PluginProps>>,
+    childOptions: ChildOptionItem[]
+): any {
+    const mappedData: any = cloneDeep(data);
+    const pluginModifiedSchemaLocation: string = mapSchemaLocationFromDataLocation(
+        pluginModifiedDataLocation.dataLocation,
+        data,
+        pluginModifiedDataLocation.schema
+    );
+    const pluginId: string = get(
+        pluginModifiedDataLocation.schema,
+        `${pluginModifiedSchemaLocation}.${pluginIdKeyword}`
+    );
+    const pluginResolver: Plugin<PluginProps> = plugins.find(
+        (plugin: Plugin<PluginProps>): boolean => {
+            return plugin.matches(pluginId);
+        }
+    );
+    const pluginData: any = get(data, pluginModifiedDataLocation.absoluteDataLocation);
+
+    if (pluginResolver !== undefined) {
+        const pluginResolverMapping: PluginResolverDataMap[] = getPluginResolverDataMap({
+            isReactChildren: pluginModifiedDataLocation.type === dataType.children,
+            pluginData,
+            pluginResolver,
+            childOptions,
+            dataLocation: pluginModifiedDataLocation.absoluteDataLocation,
+        });
 
-                set(mappedData, reactChildrenDataLocation, value);
+        pluginResolverMapping.forEach(
```

Comment:
You can also chain this off of `sort` - then you don't need to declare a variable.